crates/bevy_animation/src/lib.rs

```diff
@@ -65,6 +81,50 @@ pub struct VariableCurve {
     pub interpolation: Interpolation,
 }
 
+impl VariableCurve {
+    /// Find the index of the keyframe at or before the current time.
+    ///
+    /// Returns [`None`] if the curve is finished or not yet started.
+    /// To be more precise, this returns [`None`] if the frame is at or past the last keyframe:
+    /// we cannot get the *next* keyframe to interpolate to in that case.
+    pub fn find_current_keyframe(&self, seek_time: f32) -> Option<usize> {
+        // An Ok(keyframe_index) result means an exact result was found by binary search
+        // An Err result means the keyframe was not found, and the index is the keyframe
+        // PERF: finding the current keyframe can be optimised
+        let search_result = self
+            .keyframe_timestamps
+            .binary_search_by(|probe| probe.partial_cmp(&seek_time).unwrap());
```

Comment:
im willing to bet 2 cents and a teaspoon of maple syrup that the extremely minor perf regression is thanks to the unwrap check, and something as smoothbrained and inelegant as
`|probe| if probe < &seek_time { Ordering::Less } else if probe == &seek_time { Ordering::Equal } else { Ordering::Greater }`
would fix it

Reply:
I tried comparing with tracy, the manual compare is a tiny bit worse.

I don't want the 2 cents, but I would love that teaspoon of maple syrup!

Reply:
alright come over, im in berkeley. i'll even make you a pancake to put it on